################################################################################
SHORT DESCRIPTION: 
################################################################################
Detects XciVMM CPUID signature as XenVMM.

################################################################################
LONG DESCRIPTION: 
################################################################################
OpenXT changes CPUID instruction emulation to return "XciVMMXciVMM" as
signature. SeaBIOS still needs to detect it is running on Xen.

################################################################################
CHANGELOG 
################################################################################
Documented: Eric Chanudet, chanudete@ainfosec.com, 17/03/2015
Ported to SeaBIOS 1.7.5: Eric Chanudet, chanudete@ainfosec.com, 01/03/2015

################################################################################
REMOVAL 
################################################################################
This patch is required to handle Xen modified CPUID signature.

################################################################################
UPSTREAM PLAN 
################################################################################
This is OpenXT specific. There is no plan to upstream this patch.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################
Depends on Xen: xc-xt-cpuid.patch

################################################################################
PATCHES
################################################################################
Index: seabios-1.9.1/src/fw/xen.c
===================================================================
--- seabios-1.9.1.orig/src/fw/xen.c
+++ seabios-1.9.1/src/fw/xen.c
@@ -69,7 +69,8 @@ void xen_preinit(void)
 
         dprintf(9, "Found hypervisor signature \"%s\" at %x\n",
                 signature, base);
-        if (strcmp(signature, "XenVMMXenVMM") == 0) {
+        if (strcmp(signature, "XenVMMXenVMM") == 0 ||
+            strcmp(signature, "XciVMMXciVMM") == 0) {
             /* Set debug_io_port first, so the following messages work. */
             DebugOutputPort = 0xe9;
             debug_banner();
